src/main/java/io/javalin/core/JavalinConfig.java

```diff
@@ -172,9 +175,19 @@ public JavalinConfig configureServletContextHandler(Consumer<ServletContextHandl
         return this;
     }
 
+    public JavalinConfig configureDynamicCompressionStrategy(DynamicCompressionStrategy strategy) {
+        inner.dynamicCompressionStrategy = strategy;
+        return this;
+    }
+
     public static void applyUserConfig(Javalin app, JavalinConfig config, Consumer<JavalinConfig> userConfig) {
         userConfig.accept(config); // apply user config to the default config
 
+        //If compression strategy wasn't defined by user, create one based on the old dynamicGzip flag
+        if(config.inner.dynamicCompressionStrategy == DynamicCompressionStrategy.NONE) {
+            config.inner.dynamicCompressionStrategy = new DynamicCompressionStrategy(false, config.dynamicGzip);
+        }
```

Comment:
Check if `disableDynamicGzip` is set, then change to `NONE` ?

Reply:
When you say disableDynamicGzip, do you mean just setting the dynamicGzip boolean to false?

If I were to set the compression strategy to NONE based on that, the deprecated boolean would take priority over the compression strategy and act as a global off switch for ALL dynamic compression. This would act as a complete opposite to what it's doing now, and would give all the "power" to the deprecated boolean. Is this really what you want, or am I misunderstanding something?

The way I have it set now:
- Starts off with NONE (which is not null, just both disabled)
- Waits for user config settings to be applied
- Checks if a new DynamicCompressionStrategy was defined by the user
- If not, it falls back to the deprecated boolean and creates a strategy from that

Reply:
> Is this really what you want, or am I misunderstanding something?

Well, when you say it like that ... 😬 

But that was what I was suggesting, yes. My reasoning is that we want the code-base to conform to the new way of doing things, so we should set the default value to be `Strategy.GZIP`, and handle the old way of doing things as a special case (easy to just delete those lines when the variable is removed later).

I don't mind giving power to the deprecated boolean, because the only people who would be affected by it are the people who both have managed to set a compression strategy, but who are also for some reason using a deprecated config option. 

Their thought process would have be strange:

* I want GZIP and Brotli, so I should create a strategy: `stratey(gzip(4), brotli(6))`. 
* Actually, I don't want GZIP after all, so I should use the deprecated config option `disableGzip` (which has docs pointing to strategy)

It just seems unlikely to happen. I don't feel strongly about this though. Your way of doing it is more correct, but makes for a little more confusing code (imo).

Reply:
Sorry! I did not mean to sound abrasive there. I was just in my "max focus" mode, and I wanted to make sure that I understood correctly. :D

I actually think you make a good point. I was trying too hard to account for user silliness, but I forget this is an API for developers. They should be able to figure out mixing old and new options is a bad idea!

I am going in favour of easier code readability/maintenance, and will implement your suggestion.